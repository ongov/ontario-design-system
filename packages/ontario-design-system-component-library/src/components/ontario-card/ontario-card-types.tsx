export const Layouts = ['vertical', 'horizontal'] as const;
export const HorizontalImagePositionTypes = ['left', 'right'] as const;
export const HorizontalImageSizeTypes = ['one-fourth', 'one-third'] as const;

export type Layout = (typeof Layouts)[number];
export type HorizontalImagePositionType = (typeof HorizontalImagePositionTypes)[number];
export type HorizontalImageSizeType = (typeof HorizontalImageSizeTypes)[number];

export const HeaderColours = [
	'dark-accent',
	'light-accent',
	'light-gold',
	'light-yellow',
	'light-taupe',
	'light-green',
	'light-lime',
	'light-teal',
	'light-sky',
	'light-blue',
	'light-purple',
	'light-orange',
	'light-red',
	'light-magenta',
	'gold',
	'yellow',
	'taupe',
	'green',
	'lime',
	'teal',
	'sky',
	'blue',
	'purple',
	'orange',
	'red',
	'magenta',
];

export type HeaderColour = (typeof HeaderColours)[number];
