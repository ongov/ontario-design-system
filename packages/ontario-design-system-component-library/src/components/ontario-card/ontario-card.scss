@use 'sass:math';
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/global.variables' as globalVariables;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/spacing.variables' as spacing;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/colours.variables' as colours;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/breakpoints.variables' as breakpoints;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/2-tools/placeholder/focus.placeholders' as
	placeholders;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/font-weights.variables' as fontWeights;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/font-sizes.variables' as fontSizes;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/2-tools/functions/global.functions' as globalFunctions;
@use '@ongov/ontario-design-system-global-styles/dist/styles/scss/1-variables/typography.variables' as typography;

.ontario-card {
	box-shadow: 0rem 0.1875rem 0.5rem 0.0625rem rgba(0, 0, 0, 0.4);
	border-radius: globalVariables.$global-radius;
	margin: spacing.$spacing-0 spacing.$spacing-7 spacing.$spacing-7 spacing.$spacing-0;
	padding: spacing.$spacing-0;
	/*
	 * Required for cards to stay the same height regardless of content size.
	 * Without subtracting the margin-bottom amount the cards will be too large.
	 */
	height: calc(100% - spacing.$spacing-7);
	list-style-type: none;
	transition: all 0.3s ease-in-out;
	position: relative;
	cursor: pointer;

	&:hover {
		box-shadow: 0rem 0.375rem 0.75rem 0.125rem rgba(0, 0, 0, 0.35);
	}

	&:focus-within {
		box-shadow: placeholders.$ontario-focus-box-shadow;
		outline: 4px solid transparent;
		transition: box-shadow 0.1s ease-in-out;
	}

	&:active {
		box-shadow: placeholders.$ontario-focus-box-shadow;
		background-color: colours.$ontario-greyscale-5;
	}

	@media screen and (max-width: breakpoints.$small-breakpoint) {
		margin-bottom: spacing.$spacing-8;
		width: 100% !important; // to override the specificity of the widths listed for the card-rows
		max-width: 100%;
	}
}

.ontario-card--image-true {
	margin-top: spacing.$spacing-2 * -1;
	@at-root .ontario-card__card-type--horizontal & {
		margin-top: spacing.$spacing-0;
	}
}

.ontario-card__card-type--horizontal {
	display: flex;

	@media screen and (max-width: breakpoints.$small-breakpoint) {
		width: 100%;
		max-width: 100%;
	}

	.ontario-card__image-container,
	.ontario-card__text-container {
		position: relative;
	}

	.ontario-card__heading {
		border-radius: 0;
	}

	a {
		&:focus-within {
			box-shadow: none;
			outline: none;
		}

		&:active {
			box-shadow: none;
		}
	}

	img {
		height: 100%;
		position: absolute;
	}
}

.ontario-card__image-right {
	flex-direction: row-reverse;
}

.ontario-card__heading {
	@extend %h4-styles;
	margin: spacing.$spacing-0;
	border-radius: globalVariables.$global-radius globalVariables.$global-radius 0 0;
	padding: (spacing.$spacing-5 - spacing.$spacing-1) spacing.$spacing-5 spacing.$spacing-2 spacing.$spacing-5;
	transition: text-decoration 0.3s ease-in-out;
	background-color: colours.$ontario-colour-white;
	max-width: none;

	.ontario-card:hover & {
		text-decoration-line: underline;
		text-decoration-color: colours.$ontario-colour-black;
	}

	.ontario-card--image-true & {
		border-radius: 0;
		@at-root .ontario-card__card-type--horizontal & {
			border-radius: 0 globalVariables.$global-radius 0 0;
		}
	}

	a {
		color: colours.$ontario-colour-black;
		text-decoration: none;
		outline: none;

		&::after {
			content: '';
			position: absolute;
			left: 0;
			top: 0;
			right: 0;
			bottom: 0;
		}

		&:focus {
			box-shadow: none;
		}

		&:active {
			outline: none;
		}
	}

	@media screen and (max-width: breakpoints.$medium-breakpoint) {
		font-size: fontSizes.$ontario-font-size-large;
		padding-left: spacing.$spacing-4;
		padding-right: spacing.$spacing-4;
	}
}

// ontario-card's header colour options

$ontario-card-heading-colours: (
	'light-accent': colours.$ontario-greyscale-5,
	'dark-accent': colours.$ontario-colour-black,
	'light-gold': colours.$ontario-colour-light-gold,
	'light-yellow': colours.$ontario-colour-light-yellow,
	'light-taupe': colours.$ontario-colour-light-taupe,
	'light-green': colours.$ontario-colour-light-green,
	'light-lime': colours.$ontario-colour-light-lime,
	'light-teal': colours.$ontario-colour-light-teal,
	'light-sky': colours.$ontario-colour-light-sky,
	'light-purple': colours.$ontario-colour-light-purple,
	'light-orange': colours.$ontario-colour-light-orange,
	'light-red': colours.$ontario-colour-light-red,
	'light-magenta': colours.$ontario-colour-light-magenta,
	'gold': colours.$ontario-colour-gold,
	'yellow': colours.$ontario-colour-yellow,
	'taupe': colours.$ontario-colour-taupe,
	'green': colours.$ontario-colour-green,
	'lime': colours.$ontario-colour-lime,
	'teal': colours.$ontario-colour-teal,
	'sky': colours.$ontario-colour-sky,
	'purple': colours.$ontario-colour-purple,
	'orange': colours.$ontario-colour-orange,
	'red': colours.$ontario-colour-red,
	'magenta': colours.$ontario-colour-magenta,
);

@each $colour-name, $colour-value in $ontario-card-heading-colours {
	.ontario-card__background--#{$colour-name} {
		background-color: $colour-value;
	}

	.ontario-card__heading--#{$colour-name} {
		background-color: $colour-value;
		padding-top: spacing.$spacing-4;
		padding-bottom: spacing.$spacing-4;
	}
}

.ontario-card__text-container {
	.ontario-card__card-type--horizontal & {
		width: 66.6%;
	}

	.ontario-card__image-size-one-fourth & {
		width: 75%;
	}

	.ontario-card--no-image & {
		width: 100%;
	}
}

.ontario-card__description {
	margin: spacing.$spacing-0;
	padding: spacing.$spacing-2 spacing.$spacing-5 spacing.$spacing-5 spacing.$spacing-5;
	background-color: colours.$ontario-colour-white;
	border-bottom-left-radius: globalVariables.$global-radius;
	border-bottom-right-radius: globalVariables.$global-radius;

	p {
		margin-top: spacing.$spacing-0;
	}

	p:last-of-type {
		margin-bottom: spacing.$spacing-0;
	}

	@media screen and (max-width: breakpoints.$medium-breakpoint) {
		padding: spacing.$spacing-2 spacing.$spacing-4 spacing.$spacing-5 spacing.$spacing-4;
	}
}

.ontario-card__image-container {
	.ontario-card--position-vertical & {
		margin-bottom: spacing.$spacing-2 * -1;
	}

	.ontario-card__card-type--horizontal & {
		width: 33.3%;
	}

	.ontario-card__image-size-one-fourth & {
		width: 25%;
	}
}

.ontario-card__image {
	width: 100%;
	height: 100%;
	object-fit: cover;
	background-size: 100% 100%;

	.ontario-card--position-vertical & {
		border-top-left-radius: globalVariables.$global-radius;
		border-top-right-radius: globalVariables.$global-radius;
	}

	.ontario-card__image-right & {
		border-top-right-radius: globalVariables.$global-radius;
		border-bottom-right-radius: globalVariables.$global-radius;
	}

	.ontario-card__image-left & {
		border-top-left-radius: globalVariables.$global-radius;
		border-bottom-left-radius: globalVariables.$global-radius;
	}
}
