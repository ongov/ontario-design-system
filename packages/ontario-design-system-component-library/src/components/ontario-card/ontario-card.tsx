import { Component, Prop, Element, h, State, Watch } from '@stencil/core';
import {
	HeaderColour,
	HeaderColours,
	HorizontalImagePositionType,
	HorizontalImageSizeType,
	Layout,
	Layouts,
} from './ontario-card-types';
import { validateValueAgainstArray } from '../../utils/validation/validation-functions';
import { ConsoleMessageClass } from '../../utils/console-message/console-message';
import { printArray } from '../../utils/helper/utils';

@Component({
	tag: 'ontario-card',
	styleUrl: 'ontario-card.scss',
	shadow: true,
})
export class OntarioCard {
	@Element() host: HTMLElement;

	/**
	 * Text to be displayed within the header.
	 *
	 * @example
	 *	<ontario-card
	 *		header-type="dark"
	 *		card-type="horizontal"
	 *		label="Card Title 1"
	 *		description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
	 *	>
	 */
	@Prop() label: string;

	/**
	 * Image to be displayed within the card image container.
	 *
	 * This is optional.
	 */
	@Prop() image?: string;

	/**
	 * Text to be displayed within the card description container.
	 *
	 * This is optional.
	 */
	@Prop() description?: string;

	/**
	 * Action link for when the card is clicked.
	 *
	 * This is optional.
	 */
	@Prop() cardLink?: string;

	/**
	 * The layout oritnetation of the card.
	 *
	 * If no type is passed, it will default to 'vertical'.
	 *
	 */
	@Prop() layout?: Layout = 'vertical';

	/**
	 * Set the card's header colour.
	 *
	 * This is optional.
	 */
	@Prop() headerColour?: HeaderColour;

	/**
	 * The position of the image when the card-type is set to 'horizontal'.
	 *
	 * This prop is only necessry when the card-type is set to 'horizontal'.
	 *
	 * @example
	 * 	<ontario-card
	 *		card-type="horizontal"
	 *		label="Card Title 1"
	 *		image="https://picsum.photos/200/300"
	 *		horizontal-image-position-type="left"
	 *		horizontal-image-size-type="one-fourth"
	 *	  description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
	 *	>
	 *	</ontario-card>
	 */
	@Prop() horizontalImagePositionType?: HorizontalImagePositionType = 'left';

	/**
	 * The size of the image when the card-type is set to 'horizontal'.
	 *
	 * This prop is only necessry when the card-type is set to 'horizontal'.
	 *
	 * @example
	 * 	<ontario-card
	 *		card-type="horizontal"
	 *		label="Card Title 1"
	 *		image="https://picsum.photos/200/300"
	 *		horizontal-image-position-type="left"
	 *		horizontal-image-size-type="one-fourth"
	 *	  description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
	 *	>
	 *	</ontario-card>
	 */
	@Prop() horizontalImageSizeType?: HorizontalImageSizeType = 'one-third';

	/**
	 * Provides more context as to what the card interaction is doing. This should only be used for accessibility purposes, if the card interaction requires more * * description than what the text provides.
	 *
	 * This is optional.
	 *
	 */
	@Prop() ariaLabelText?: string;

	/**
	 * Mutable variable, for internal use only.
	 * Set the card's layout depending on validation result.
	 */
	@State() private layoutState: string;

	/**
	 * Watch for changes to the `layout` property for validation purposes.
	 *
	 * If the user input doesn't match one of the array values then `layout` will be set to its default (`vertical`).
	 * If a match is found in one of the array values then `layoutState` will be set to the matching array key value.
	 */
	@Watch('layout')
	validateLayout() {
		if (this.layout) {
			const isValid = validateValueAgainstArray(this.layout, Layouts);
			if (isValid) {
				this.layoutState = this.layout;
			} else {
				this.warnDefaultLayout();
				this.layoutState = 'vertical';
			}
		}
	}

	/**
	 * Watch for changes to the `headerColour` property for validation purposes.
	 *
	 * If the user input doesn't match one of the array values then `headerColour` will be kept empty ('').
	 * If a match is found in one of the array values then `headerColour` will be set to the matching array key value.
	 */
	@Watch('headerColour')
	validateHeaderColour() {
		if (this.headerColour) {
			const isValid = validateValueAgainstArray(this.headerColour, HeaderColours);

			if (!isValid) {
				this.warnDefaultHeaderColour();
				this.headerColour = '';
			}
		}
	}

	/**
	 * Print the invalid `layout` prop warning message.
	 */
	private warnDefaultLayout() {
		const message = new ConsoleMessageClass();
		message
			.addDesignSystemTag()
			.addMonospaceText(' layout ')
			.addRegularText('on')
			.addMonospaceText(' <ontario-card> ')
			.addRegularText('was set to an invalid layout; only ')
			.addMonospaceText(printArray([...Layouts]))
			.addRegularText(' are supported. The default layout')
			.addMonospaceText(' vertical ')
			.addRegularText('is assumed.')
			.printMessage();
	}

	/**
	 * Print the invalid `headerColour` prop warning message.
	 */
	private warnDefaultHeaderColour() {
		const message = new ConsoleMessageClass();
		message
			.addDesignSystemTag()
			.addMonospaceText(' header-colour ')
			.addRegularText('on')
			.addMonospaceText(' <ontario-card> ')
			.addRegularText('was set to an invalid colour; only ')
			.addMonospaceText(printArray([...HeaderColours]))
			.addRegularText(' are supported. ')
			.addRegularText('No colour is assumed as the default.')
			.printMessage();
	}

	/**
	 * Returns the top level classes of the card.
	 *
	 * @returns {string}
	 */
	private getCardClasses(): string {
		const baseClass =
			this.layoutState === 'horizontal'
				? `ontario-card ontario-card__card-type--${this.layoutState} ontario-card__image-${this.horizontalImagePositionType} ontario-card__image-size-${this.horizontalImageSizeType}`
				: `ontario-card ontario-card__card-type--basic ontario-card--position-${this.layoutState}`;

		const descriptionClass = this.description ? '' : ' ontario-card__description-false';

		const backgroundClass =
			this.headerColour && !this.description ? `ontario-card__background--${this.headerColour}` : '';

		return `${baseClass} ${descriptionClass} ${backgroundClass}`.trim();
	}

	/**
	 * Returns the heading classes of the card.
	 *
	 * @returns {string}
	 */
	private getCardHeadingClasses(): string {
		const baseClass = 'ontario-card__heading';

		const backgroundClass =
			this.headerColour && validateValueAgainstArray(this.headerColour, HeaderColours)
				? `ontario-card__heading--${this.headerColour}`
				: '';

		return `${baseClass} ${backgroundClass}`.trim();
	}

	/**
	 * Returns the url that the card links to.
	 *
	 * @returns {string}
	 */
	private getHref(): string {
		return this.cardLink ? this.cardLink : '#';
	}

	/**
	 * Component life cycle hook.
	 *
	 * https://stenciljs.com/docs/component-lifecycle#connectedcallback
	 */
	componentWillLoad() {
		this.validateLayout();
		this.validateHeaderColour();
	}

	render() {
		return (
			<li class={this.getCardClasses()}>
				{this.image && (
					<div class="ontario-card__image-container">
						<a href={this.getHref()} aria-label={this.ariaLabelText}>
							<img class="ontario-card__image" src={this.image} />
						</a>
					</div>
				)}
				<div class={`ontario-card__text-container ${this.image ? 'ontario-card--image-true' : ''}`}>
					<h2 class={this.getCardHeadingClasses()}>
						<a href={this.getHref()} aria-label={this.ariaLabelText}>
							{this.label}
						</a>
					</h2>
					{this.description && (
						<div class="ontario-card__description">
							<p>{this.description}</p>
						</div>
					)}
				</div>
			</li>
		);
	}
}
