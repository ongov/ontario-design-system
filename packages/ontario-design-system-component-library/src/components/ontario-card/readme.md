import { OntarioCard } from '@ongov/ontario-design-system-component-library-react';

# ontario-card

Use a card to provide an entry point to information on a different page.

## Usage guidance

Please refer to the [Ontario Design System](https://designsystem.ontario.ca/components/detail/cards.html) for current documentation guidance. This component needs to be used in conjunction with the `ontario-card-collection` component. Please reference the [`ontario-card-collection`](../ontario-card-collection/) readme for further details.

## Configuration

Once the component package has been installed (see Ontario Design System Component Library for installation instructions), the `ontario-card` component can be added directly into the project's code, and can be customized by updating the properties outlined [here](#properties). Please see the [examples](#examples) below for how to configure the component.

## Examples

Example of a bare-bones `ontario-card` component where the `label` and `description` for the card is provided.

```html
<ontario-card label="Card Title" description="Lorem Ipsum Lorem Ipsum"></ontario-card>
```

<OntarioCard label="Card Title" description="Lorem Ipsum Lorem Ipsum"></OntarioCard>

Example of an `ontario-card` component with an image, which includes the `headerColour`, `image`, `label` and `description` properties.

```html
<ontario-card
	label="Card Title"
	header-colour="sky"
	image="https://picsum.photos/300/225"
	description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
>
</ontario-card>
```

<div>
	<OntarioCard
		label="Card Title"
		headerColour="sky"
		image="https://picsum.photos/300/225"
		description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
	></OntarioCard>
</div>

This is another example of an `ontario-card` component with a horizontal layout containing an image. This will require more properties, including the `horizontalImagePositionType` and the `horizontalImageSizeType`.

```html
<ontario-card
	layout="horizontal"
	label="Card Title"
	image="https://picsum.photos/300/225"
	horizontal-image-position-type="left"
	horizontal-image-size-type="one-fourth"
	description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
>
</ontario-card>
```

<div>
	<OntarioCard
		layout="horizontal"
		label="Card Title"
		image="https://picsum.photos/300/225"
		horizontalImagePositionType="left"
		horizontalImageSizeType="one-fourth"
		description="Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum"
	></OntarioCard>
</div>

## Custom property types

### headerColour

The `header-colour` property supports a wide range of values, including:

- `dark-accent`
- `light-accent`
- `light-gold`
- `light-yellow`
- `light-taupe`
- `light-green`
- `light-lime`
- `light-teal`
- `light-sky`
- `light-blue`
- `light-purple`
- `light-orange`
- `light-red`
- `light-magenta`
- `gold`
- `yellow`
- `taupe`
- `green`
- `lime`
- `teal`
- `sky`
- `blue`
- `purple`
- `orange`
- `red`
- `magenta`

<!-- Auto Generated Below -->

## Properties

| Property                      | Attribute                        | Description                                                                                                                                                                                                                 | Type                                       | Default       |
| ----------------------------- | -------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------ | ------------- |
| `ariaLabelText`               | `aria-label-text`                | Provides more context as to what the card interaction is doing. This should only be used for accessibility purposes, if the card interaction requires more \* \* description than what the text provides. This is optional. | `string \| undefined`                      | `undefined`   |
| `cardLink`                    | `card-link`                      | Action link for when the card is clicked. This is optional.                                                                                                                                                                 | `string \| undefined`                      | `undefined`   |
| `description`                 | `description`                    | Text to be displayed within the card description container. This is optional.                                                                                                                                               | `string \| undefined`                      | `undefined`   |
| `headerColour`                | `header-colour`                  | Set the card's header colour. This is optional.                                                                                                                                                                             | `string \| undefined`                      | `undefined`   |
| `horizontalImagePositionType` | `horizontal-image-position-type` | The position of the image when the card-type is set to 'horizontal'. This prop is only necessry when the card-type is set to 'horizontal'.                                                                                  | `"left" \| "right" \| undefined`           | `'left'`      |
| `horizontalImageSizeType`     | `horizontal-image-size-type`     | The size of the image when the card-type is set to 'horizontal'. This prop is only necessry when the card-type is set to 'horizontal'.                                                                                      | `"one-fourth" \| "one-third" \| undefined` | `'one-third'` |
| `image`                       | `image`                          | Image to be displayed within the card image container. This is optional.                                                                                                                                                    | `string \| undefined`                      | `undefined`   |
| `label`                       | `label`                          | Text to be displayed within the header.                                                                                                                                                                                     | `string`                                   | `undefined`   |
| `layout`                      | `layout`                         | The layout oritnetation of the card. If no type is passed, it will default to 'vertical'.                                                                                                                                   | `"horizontal" \| "vertical" \| undefined`  | `'vertical'`  |

---

_Built with [StencilJS](https://stenciljs.com/)_
