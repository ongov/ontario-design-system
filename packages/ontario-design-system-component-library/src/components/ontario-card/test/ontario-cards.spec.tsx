import { newSpecPage } from '@stencil/core/testing';
import { OntarioCard } from '../ontario-card';

describe('ontario-card', () => {
	it('renders', async () => {
		const page = await newSpecPage({
			components: [OntarioCard],
			html: `<ontario-card></ontario-card>`,
		});
		expect(page.root).toMatchSnapshot();
	});

	it('should render a label', async () => {
		const page = await newSpecPage({
			components: [OntarioCard],
			html: `<ontario-card label="Card Title 1"></ontario-card>`,
		});
		expect(page.root).toMatchSnapshot();
	});

	it('should render a header colour on the label', async () => {
		const page = await newSpecPage({
			components: [OntarioCard],
			html: `<ontario-card label="Card Title 1" header-colour="sky"></ontario-card>`,
		});
		expect(page.root).toMatchSnapshot();
	});

	it('should render a description', async () => {
		const page = await newSpecPage({
			components: [OntarioCard],
			html: `<ontario-card description="Lorem ipsum"></ontario-card>`,
		});
		expect(page.root).toMatchSnapshot();
	});

	// Don't think we can test images unless we point to a local path
});
