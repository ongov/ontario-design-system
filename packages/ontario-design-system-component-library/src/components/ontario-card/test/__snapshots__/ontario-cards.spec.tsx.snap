// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ontario-card renders 1`] = `
<ontario-card>
  <mock:shadow-root>
    <li class="ontario-card ontario-card--position-vertical ontario-card__card-type--basic ontario-card__description-false">
      <div class="ontario-card__text-container">
        <h2 class="ontario-card__heading">
          <a href="#"></a>
        </h2>
      </div>
    </li>
  </mock:shadow-root>
</ontario-card>
`;

exports[`ontario-card should render a description 1`] = `
<ontario-card description="Lorem ipsum">
  <mock:shadow-root>
    <li class="ontario-card ontario-card--position-vertical ontario-card__card-type--basic">
      <div class="ontario-card__text-container">
        <h2 class="ontario-card__heading">
          <a href="#"></a>
        </h2>
        <div class="ontario-card__description">
          <p>
            Lorem ipsum
          </p>
        </div>
      </div>
    </li>
  </mock:shadow-root>
</ontario-card>
`;

exports[`ontario-card should render a header colour on the label 1`] = `
<ontario-card header-colour="sky" label="Card Title 1">
  <mock:shadow-root>
    <li class="ontario-card ontario-card--position-vertical ontario-card__background--sky ontario-card__card-type--basic ontario-card__description-false">
      <div class="ontario-card__text-container">
        <h2 class="ontario-card__heading ontario-card__heading--sky">
          <a href="#">
            Card Title 1
          </a>
        </h2>
      </div>
    </li>
  </mock:shadow-root>
</ontario-card>
`;

exports[`ontario-card should render a label 1`] = `
<ontario-card label="Card Title 1">
  <mock:shadow-root>
    <li class="ontario-card ontario-card--position-vertical ontario-card__card-type--basic ontario-card__description-false">
      <div class="ontario-card__text-container">
        <h2 class="ontario-card__heading">
          <a href="#">
            Card Title 1
          </a>
        </h2>
      </div>
    </li>
  </mock:shadow-root>
</ontario-card>
`;
